elastic/logstash
# Quotes java executable path in case it contains spaces

## Release notes

[rn:skip]

## What does this PR do?


Quotes all expansions of `JAVACMD` variable in Windows batch scripts.
This is necessary in case the path to the java executable contains spaces like in:
```
c:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot
```
 

## Why is it important/What is the impact to the user?

Enable the Windows user to use spaced paths to the JDK

## Checklist




## Author's Checklist


- [x] set a spaced LS_JAVA_HOME and check Logstash `bin\logstash` is able to run.

## How to test this PR locally


- configure a JDK in path with a spaced name like 
```
c:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot
```
this should be default path when download and install Windows' Temurin installer.
- set `LS_JAVA_HOME` to the above mentioned JVM
```
set "LS_JAVA_HOME=c:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot"
```
-  run Logstash 

## Related issues


- 

## Use cases

As a user I want to be able to run Logstash configuring a JVM located in  a spaced path. 


## Logs
Error before this patch:
```
λ bin\logstash -e "input {stdin{}} output {stdout {codec => rubydebug}}"
Using LS_JAVA_HOME defined java: c:\Program Files\AdoptOpenJDK\jdk-11.0.11.9-hotspot
"WARNING: Using LS_JAVA_HOME while Logstash distribution comes with a bundled JDK."
The system cannot find the path specified.
could not find java; set JAVA_HOME or ensure java is in PATH
```
